content/entry-hash/index.md

```diff
@@ -0,0 +1,176 @@
+---
+rfc:
+start_date: 2018-08-02
+pr:
+status: draft
+---
+
+# Entry hash
+
+## Summary
+
+This RFC proposes an algorithm to hash entries without relying in JSON or in
+the attribute names.
+
+**This RFC is incompatible with the reference implementation**
+
+## Motivation
+
+The current specfication doesn't define what is the algorithm to get from the
+entry data to the entry hash for the given hashing algorithm (from now on
+SHA-256). The reference implementation (ORJ) reuses the item canonicalisation
+algorithm to canonicalise the entry JSON representation and then hash the
+result. Finally, relying on the JSON representation has the downside of being
+more rigid than it could be.
+
+## Explanation
+
+The algorithm has three parts:
+
+1. tag and hash each value in the entry.
+2. order the resulting hashes.
+3. hash the result of concatenating all hashes.
```

Comment:
umm no, this is the algorithm for the entry hash, the item hash is different. I'm working on another RFC for that one due the redactable stream of work but they are different.

The entry has a fixed structure and it is known by the register, the item has a flexible structure that changes from register to register and when we have schema evolution it will change over time as well.

Reply:
In other words, if the item changes, its hash changes and, because the entry depends on the item hash to generate its own hash, it will change as well.